spec.html

```diff
@@ -32544,67 +32544,55 @@ <h1>Array.prototype.some ( _callbackfn_ [ , _thisArg_ ] )</h1>
+                1. Set _itemCount_ to _itemCount_ + 1.
+            1. Set _k_ to _k_ + 1.
+          1. Sort _items_ using an implementation-dependant sequence of calls to SortCompare.
+          1. Let _j_ be 0.
+          1. Repeat, while _j_ &lt; _itemCount_
+            1. Perform ? Set(_obj_, ! ToString(_j_), items[_j_], *true*).
+            1. Set _j_ to _j_ + 1.
+          1. Repeat, while _j_ &lt; _itemCount_ + _undefinedCount_
+            1. Perform ? Set(_obj_, ! ToString(_j_), *undefined*, *true*).
+            1. Set _j_ to _j_ + 1.
+          1. Repeat, while _j_ &lt; len
+            1. Let _Pj_ be ! ToString(_j_).
+            1. Let _jPresent_ be ? HasOwnProperty(_obj_, _Pj_).
+            1. If _jPresent_ is *true*, then
+              1. Perform ? DeletePropertyOrThrow(_obj_, _Pj_).
+            1. Set _j_ to _j_ + 1.
+          1. Return _obj_.
         </emu-alg>
         <p>The <em>sort order</em> is the ordering, after completion of this function, of the <emu-xref href="#integer-index">integer-indexed</emu-xref> property values of _obj_ whose integer indexes are less than _len_. The result of the `sort` function is then determined as follows:</p>
-        <p>If _comparefn_ is not *undefined* and is not a consistent comparison function for the elements of this array (see below), the sort order is implementation-defined. The sort order is also implementation-defined if _comparefn_ is *undefined* and SortCompare does not act as a consistent comparison function.</p>
-        <p>Let _proto_ be _obj_.[[GetPrototypeOf]](). If _proto_ is not *null* and there exists an integer _j_ such that all of the conditions below are satisfied then the sort order is implementation-defined:</p>
+        <p>The sort order is implementation-defined if any of the following conditions are true:</p>
         <ul>
           <li>
-            _obj_ is sparse
-          </li>
-          <li>
-            0 &le; _j_ &lt; _len_
+            If _comparefn_ is not *undefined* and is not a consistent comparison function for the elements of _items_ (see below).
           </li>
           <li>
-            HasProperty(_proto_, ToString(_j_)) is *true*.
-          </li>
-        </ul>
-        <p>The sort order is also implementation-defined if _obj_ is sparse and any of the following conditions are true:</p>
-        <ul>
-          <li>
-            IsExtensible(_obj_) is *false*.
-          </li>
-          <li>
-            Any integer index property of _obj_ whose name is a nonnegative integer less than _len_ is a data property whose [[Configurable]] attribute is *false*.
-          </li>
-        </ul>
-        <p>The sort order is also implementation-defined if any of the following conditions are true:</p>
-        <ul>
-          <li>
-            If _obj_ is an exotic object (including Proxy exotic objects) whose behaviour for [[Get]], [[Set]], [[Delete]], and [[GetOwnProperty]] is not the ordinary object implementation of these internal methods.
```

Comment:
Same as above, these are invoked in the same order independent of the concrete sorting algorithm used.